library/test/src/lib.rs

```diff
@@ -591,6 +593,28 @@ fn run_test_in_process(
     // Buffer for capturing standard I/O
     let data = Arc::new(Mutex::new(Vec::new()));
 
+    // If we encounter a non-unwinding panic, stop capturing output to ensure that the
+    // non-unwinding panic message is visible.
+    // We also acquire the locks for both output streams to prevent output from other threads
+    // from interleaving with the panic message or appearing after it.
+    let builtin_panic_hook = panic::take_hook();
+    let hook = Box::new({
+        let data = data.clone();
+        move |info: &'_ PanicInfo<'_>| {
+            if !info.can_unwind() {
+                io::set_output_capture(None);
+                let _ = ManuallyDrop::new(crate::io::stderr().lock());
```

Comment:
Sorry for the back and forth, but I only meant for stdout. This line is much clearer as `core::mem::forget(crate::io::stderr().lock());`

Reply:
Thanks!